### ISSUE TITLE
EXTRA_ENCODINGS of ACTION_HDMI_AUDIO_PLUG intent of AudioManager reports EAC3 even when user has selected PCM only in Audio Output

### ISSUE THREAD

**Issue by PratimMallick** [google/ExoPlayer#7669]
Using a TCL AndroidTV to play dolby digital stream(Ec3) and an external 5.1 is connected to the TV via HDMI ARC. 

While deciding the AudioCapabilities we are reading the `EXTRA_ENCODINGS` of `ACTION_HDMI_AUDIO_PLUG` intent of audioManager. This reports the correct value of `ENCODING_E_AC3` when Auto or Passthrough option is selected in the android tv's 'Digital Audio Out' user settings. 
But when PCM is selected in this setting, the `EXTRA_ENCODINGS` still give a value of `ENCODING_E_AC3` and `maxChannelCount` also comes to be 6 suggesting passthrough is possible.

Is this expected or a bug in Android framework? How does exoplayer detect the user setting correctly? Is there any Key that should be read by exoplayer?

**Comment by ojw28** [comment]
@andrewlewis - I don't really understand why this isn't just a duplicate of https://github.com/google/ExoPlayer/issues/7617 (if it is, please close it as such; thanks!).

**Comment by andrewlewis** [comment]
If I understand correctly, #7617 is actually about detecting optical plug events, while this issue is about AC3 output unexpectedly being reported as possible when the user overrides the encoded output setting to 'PCM only'.

Please could you check whether AC3 output actually works while in this PCM-only mode? It may be the case that the device supports transcoding AC3 down to PCM before sending it over HDMI.

**Comment by PratimMallick** [comment]
Yes @andrewlewis EC3 output works even if PCM-only mode is selected.
